stacktrace.go

```diff
@@ -20,23 +20,75 @@
 
 package zap
 
-import "runtime"
+import (
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap/internal/bufferpool"
+)
+
+var (
+	_stacktraceIgnorePrefixes = []string{
+		"go.uber.org/zap",
+		"runtime.",
+	}
+	_stacktraceUintptrPool = sync.Pool{
+		New: func() interface{} {
+			return make([]uintptr, 64)
+		},
+	}
```

Comment:
We shouldn't directly pool `[]uintptr`. Slices are actually values (see "Slices: The slice header" in [this Go Blog post](https://blog.golang.org/slices)), so passing them as `interface{}` allocates. Since `sync.Pool.Put` takes an `interface{}`, putting a plain `[]uintptr` back into the pool allocates, which is exactly what we're trying to avoid.

Instead, try this:

```
type programCounters struct {
  pcs []uintptr
}
```

and pool `*programCounters`.